File: SerenePSM/SerenePSM.Web/Modules/Default/Pfk7171/Pfk7171Columns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace SerenePSM.Default.Columns;

[ColumnsScript("Default.Pfk7171")]
[BasedOnRow(typeof(Pfk7171Row), CheckNames = true)]
public class Pfk7171Columns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]

    [Width(100), QuickFilter]

    
    public string BasicCode { get; set; }

    public string K7171_BasicSel { get; set; }

    public string BasicName { get; set; }

    
    public string NameHlpButton { get; set; }
    public decimal DisplaySeq { get; set; }
    public string DevelopmentCode { get; set; }
    public string SeBasicMaster { get; set; }
    public string CdBasicMaster { get; set; }


    public string NameSimply { get; set; }
    public string ForeignName1 { get; set; }
    public string ForeignName2 { get; set; }
    public string Check1 { get; set; }
    public string Check2 { get; set; }
    public string Check3 { get; set; }
    public string Check4 { get; set; }
    public string Check5 { get; set; }
    public string Check6 { get; set; }
    public string Check7 { get; set; }
    public string Check8 { get; set; }
    public string Check9 { get; set; }
    public string Check10 { get; set; }
    public string CheckName1 { get; set; }
    public string CheckName2 { get; set; }
    public string CheckName3 { get; set; }
    public string CheckName4 { get; set; }
    public string CheckName5 { get; set; }
    public string CheckName6 { get; set; }
    public string CheckName7 { get; set; }
    public string CheckName8 { get; set; }
    public string CheckName9 { get; set; }
    public string CheckName10 { get; set; }
    public string InchargeInsert { get; set; }
    public string InchargeUpdate { get; set; }
    public string DateInsert { get; set; }
    public string DateUpdate { get; set; }
    public string TimeInsert { get; set; }
    public string TimeUpdate { get; set; }
    public string TimeLast { get; set; }
    public string CheckSync { get; set; }
    public string CheckUse { get; set; }
    public string TableAcc { get; set; }
    public string TypeDm { get; set; }
    public string KeyDm { get; set; }
    public string ValueDm { get; set; }
    public string Remark { get; set; }
    public string SeSite { get; set; }
    public string CdSite { get; set; }
}

File: SerenePSM/SerenePSM.Web/Modules/ServerTypes/Default/Pfk7171Columns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { Pfk7171Row } from "./Pfk7171Row";

export interface Pfk7171Columns {
    BasicSel: Column<Pfk7171Row>;
    BasicCode: Column<Pfk7171Row>;
    BasicName: Column<Pfk7171Row>;
    NameHlpButton: Column<Pfk7171Row>;
    DisplaySeq: Column<Pfk7171Row>;
    DevelopmentCode: Column<Pfk7171Row>;
    SeBasicMaster: Column<Pfk7171Row>;
    CdBasicMaster: Column<Pfk7171Row>;
    NameSimply: Column<Pfk7171Row>;
    ForeignName1: Column<Pfk7171Row>;
    ForeignName2: Column<Pfk7171Row>;
    Check1: Column<Pfk7171Row>;
    Check2: Column<Pfk7171Row>;
    Check3: Column<Pfk7171Row>;
    Check4: Column<Pfk7171Row>;
    Check5: Column<Pfk7171Row>;
    Check6: Column<Pfk7171Row>;
    Check7: Column<Pfk7171Row>;
    Check8: Column<Pfk7171Row>;
    Check9: Column<Pfk7171Row>;
    Check10: Column<Pfk7171Row>;
    CheckName1: Column<Pfk7171Row>;
    CheckName2: Column<Pfk7171Row>;
    CheckName3: Column<Pfk7171Row>;
    CheckName4: Column<Pfk7171Row>;
    CheckName5: Column<Pfk7171Row>;
    CheckName6: Column<Pfk7171Row>;
    CheckName7: Column<Pfk7171Row>;
    CheckName8: Column<Pfk7171Row>;
    CheckName9: Column<Pfk7171Row>;
    CheckName10: Column<Pfk7171Row>;
    InchargeInsert: Column<Pfk7171Row>;
    InchargeUpdate: Column<Pfk7171Row>;
    DateInsert: Column<Pfk7171Row>;
    DateUpdate: Column<Pfk7171Row>;
    TimeInsert: Column<Pfk7171Row>;
    TimeUpdate: Column<Pfk7171Row>;
    TimeLast: Column<Pfk7171Row>;
    CheckSync: Column<Pfk7171Row>;
    CheckUse: Column<Pfk7171Row>;
    TableAcc: Column<Pfk7171Row>;
    TypeDm: Column<Pfk7171Row>;
    KeyDm: Column<Pfk7171Row>;
    ValueDm: Column<Pfk7171Row>;
    Remark: Column<Pfk7171Row>;
    SeSite: Column<Pfk7171Row>;
    CdSite: Column<Pfk7171Row>;
}

export class Pfk7171Columns extends ColumnsBase<Pfk7171Row> {
    static readonly columnsKey = 'Default.Pfk7171';
    static readonly Fields = fieldsProxy<Pfk7171Columns>();
}